### PR TITLE
[Inductor] Add decomposition for prims.uniform

### PR DESCRIPTION
**Pull request by lgeiger**
Currently inductor uses an eager fallback for `aten.uniform` and `aten.uniform_`. This PR adds a straight forward decomposition of `prims.uniform` so it can be natively lowered to `prims.rand`.

/cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @eellison, @shunting314, @jansel, @eikanwang

### CODE REVIEW

**Comment by jansel** [comment]
@claude explain those CI failures, check if they are related to this PR, and review the changes

**Comment by lgeiger** [comment]
Fixed the CI error and fixed the stride handling for the fast path.

File: test/inductor/test_torchinductor_opinfo.py [hunk 1]
```diff
@@ -420,6 +421,7 @@ def wrapper_noop_set_seed(op, *args, **kwargs):
     "empty_strided": {"assert_equal": False},
     "new_empty_strided": {"assert_equal": False},
     "randn": {"assert_equal": False},
+    "nn.functional.rrelu": {"check_gradient": False},
```
Comment (jansel) [inline test/inductor/test_torchinductor_opinfo.py:415]: Is this a regression?  What is going on here?

**Review by jansel — CHANGES_REQUESTED** [review]

File: test/inductor/test_torchinductor_opinfo.py (hunk 1 above)
Comment (lgeiger) [reply test/inductor/test_torchinductor_opinfo.py:415]: As far as I can tell, it’s not a regression. It's expected because `rrelu` uses `uniform` internally and `uniform` is now compiled.
Previously, `uniform` fell back to eager mode, so Inductor and Eager would generate the same random noise mask. Now that `uniform` is compiled it uses Inductor's Triton Philox RNG, which differs from eager mode. The numerical forward pass checks are already [automatically disabled for operations that include RNG ops](https://github.com/pytorch/pytorch/blob/c004e2dc8082e016df1893e3f42a831ea3f12c75/test/inductor/test_torchinductor_opinfo.py#L1393-L1407) but the gradient checks still run which causes the tests to fail.

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

File: torch/_inductor/lowering.py [hunk 2]
```diff
@@ -3623,7 +3623,7 @@ def is_aligned(x):
 
 
 # 1) Easy
-make_fallback(aten.uniform, warn=False)
+make_fallback(aten.uniform, override_decomp=True)
```
Comment (jansel) [inline torch/_inductor/lowering.py:3636]: Why override_decomp?

Comment (lgeiger) [reply torch/_inductor/lowering.py:3636]: > Why override_decomp?

@jansel Good question. The way I understand it is the following: `aten.uniform` has a decomposition that's active by default, but we still need the fallback registered for the `fallback_random=True` case where the decomposition is skipped.
I think `override_decomp=True` just silences the CI guard that otherwise errors on having both a decomposition and a fallback for the same op. This basically follows what's done for `rand_like`, `randn_like`, and `randint_like`:
https://github.com/pytorch/pytorch/blob/6b48dff080d500fa04f27096c565e4822b82fde4/torch/_inductor/lowering.py#L2975-L2977

**Review by jansel — APPROVED** [review]
